### PR TITLE
Better performance with Postgres

### PR DESCRIPTION
**Pull request by giovannibonetti**
Hi! Thank you for this very useful gem.

A while ago I have forked it to improve performance, at least in my scenario with Postgres. If there is interest from someone else we could work something out to bring it to the your library.

So, my question is:
- Are you interested in this feature?
- If you are interested, do you have a suggestion about how to detect if the database is Postgres and in that case use this alternate implementation?

At least in my machine, this patch made a dramatic speed-up to the conversion (more than 10x) when serializing hundreds/thousands of records. I guess the Ruby overhead is greatly reduced using this method, and now the serialization speed depends almost exclusively on Postgres.

### CODE REVIEW

File: lib/pluck_to_hash.rb [hunk 1]
```diff
@@ -5,19 +5,12 @@ module PluckToHash
...
-          k
-        end
-      end
-
-      pluck(*keys).map do |row|
-        row = [row] if keys.size == 1
-        Hash[formatted_keys.zip(row)]
-      end
+      # http://stackoverflow.com/questions/25331778/getting-typed-results-from-activerecord-raw-sql#answer-30948357
+      @type_map ||= PG::BasicTypeMapForResults.new(connection.raw_connection)
```
Comment (khiav223577) [inline lib/pluck_to_hash.rb:9]: you should check if PG is defined or not before using it.

**Comment by khiav223577** [comment]
10x performance is awesome!
Could you show the benchmark of it?

**Comment by girishso** [comment]
Thanks @giovannibonetti for the PR!

**Comment by giovannibonetti** [comment]
Hi, @girishso 
Sorry for not replying in a while, my personal life has drained all my spare time in the last few months - marriage, moving to a new home and things like that.

In retrospect I think the 10x performance improvement was against plain old ActiveRecord objects. Probably the improvement against the prior version of the gem was far more modest.

Thank you for taking your time to making it compatible with scenarios where other databases are used and also writing the specs. 😀

**Comment by girishso** [comment]
Oh no... never mind.. LOL!

@giovannibonetti all the best for everything! :)

**Pull request merged**